src/openrct2/interface/InteractiveConsole.cpp

```diff
@@ -938,6 +942,34 @@ static int32_t cc_set(InteractiveConsole& console, const utf8** argv, int32_t ar
             }
             console.Execute("get cheat_disable_support_limits");
         }
+        else if (strcmp(argv[0], "current_rotation") == 0 && invalidArguments(&invalidArgs, int_valid[0]))
+        {
+            uint8_t currentRotation = get_current_rotation();
+            rct_window* mainWindow = window_get_main();
+            // as there are only 4 possible rotations, make any input go to 0-3 to avoid unneccessary work
+            int32_t newRotation = int_val[0] % 4;
+            if (newRotation != currentRotation && mainWindow != nullptr)
+            {
+                // if desired rotation is less than current, move counter-clockwise
+                if (newRotation < currentRotation)
+                {
+                    while (newRotation != currentRotation)
+                    {
+                        window_rotate_camera(mainWindow, -1);
+                        newRotation++;
```

Comment:
I'm confused, why is this done in a loop here rather than simply passing the direction user wanted?

Reply:
window_rotate_camera only takes a 1 or -1 to turn clockwise or counterclockwise. I was assuming that was the best function to use and so I'd have to call it multiple times.

Reply:
If it is indeed like so, then that case that's a bug in this function's design. It should take an `enum class` containing forward and backward enumeration value and there should be another function setting the rotation directly.

Reply:
`window_rotate_camera` accepts a rotation value, I'm not sure where you got the part "only takes a 1 or -1 to turn clockwise or counterclockwise" from.

Reply:
> `window_rotate_camera` accepts a rotation value, I'm not sure where you got the part "only takes a 1 or -1 to turn clockwise or counterclockwise" from.

probably from here? https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/openrct2/interface/Window.cpp#L951

Reply:
Yes, I was reading the comment above it and taking it maybe too literally as well as not thinking hard enough - the function itself works fine and as expected with numbers other than 1 and -1 (ie -2 is 2 steps counterclockwise).

So I could just replace the loops with `window_rotate_camera(mainWindow, currentRotation - newRotation);` which I've tested and works fine and should be fine once I've fixed the other issue above. Sorry for dragging this out, I appreciate the help.